packages/driver/src/cy/snapshots.ts

```diff
@@ -212,11 +237,22 @@ export const create = ($$, state) => {
 
       const snapshot = {
         name,
-        htmlAttrs,
+        htmlAttrs: $htmlAttrs,
         body,
       }
 
-      snapshotsMap.set(snapshot, { headStyleIds, bodyStyleIds })
+      if (headStyleIds && bodyStyleIds) {
+        snapshotsMap.set(snapshot, { headStyleIds, bodyStyleIds })
+      }
+
+      if (preprocessedSnapshot) {
+        // snapshot is being reified from cross origin. get inline styles of reified snapshot
+        const { headStyles, bodyStyles } = preprocessedSnapshot.styles
```

Comment:
I don't think so since we need the full snapshot as a key into the cssMap. But I did look at the code and realized I have `snapshotsMap.set(snapshot, { headStyleIds, bodyStyleIds })` twice in there 🤦 . I took a stab at trying to address the snapshot style code in [838e091](https://github.com/cypress-io/cypress/pull/20949/commits/838e0918756a05091afe95287b3475b3f26c8953). Does that look any cleaner?